src/test/java/com/google/neighborgood/IntegrationTest.java

```diff
@@ -0,0 +1,1065 @@
+
+  @BeforeClass
+  /**
+   * Sets up test class by initializing driver, wait, js executor, and clearing all datastore
+   * entities
+   */
+  public static void setupClass() {
+    WebDriverManager.chromedriver().setup();
+    driver = new ChromeDriver();
+    // FluentWait set to timeout after 60 seconds of waiting for a WebElement to be returned and
+    // polling every second
+    wait =
+        new FluentWait<WebDriver>(driver)
+            .withTimeout(Duration.ofSeconds(60))
+            .pollingEvery(Duration.ofSeconds(1))
+            .ignoring(NoSuchElementException.class);
+
+    js = (JavascriptExecutor) driver;
+    clearAllDatastoreEntities(driver);
+  }
+
+  @AfterClass
+  public static void teardown() {
+    if (driver != null) {
+      driver.quit();
+    }
+  }
+
+  @Test
+  /**
+   * Tests functionality of a guest user who logs in for the first time in the site and has to input
+   * their new user info
+   */
+  public void _01_Homepage_AsNewGuestUser_LoginAndInputUserInfo() {
+    driver.navigate().to("http://localhost:8080/");
+    ifLaggingThenRefresh();
+    String loginElement =
+        wait.until(
+                new Function<WebDriver, WebElement>() {
+                  public WebElement apply(WebDriver driver) {
+                    return driver.findElement(By.id("loginLogoutMessage"));
+                  }
+                })
+            .getText();
+
+    // Guest user should expect to see login message
+    assertEquals(
+        "Homepage login message as guest user", "Login to help out a neighbor!", loginElement);
+
+    By addTaskButton = By.id("addtaskbutton");
+    driver.manage().timeouts().implicitlyWait(5, TimeUnit.SECONDS);
+    List<WebElement> addTaskButtonElement = driver.findElements(addTaskButton);
+    // Add task button should be missing when user is not logged in
+    assertTrue(
+        "Add task button must not be present for guest users", addTaskButtonElement.isEmpty());
+
+    By dashboardIcon = By.className("dashboard-icon");
+    driver.manage().timeouts().implicitlyWait(5, TimeUnit.SECONDS);
+    List<WebElement> dashboardIconsElement = driver.findElements(dashboardIcon);
+    // Dashboard icons should be missing when user is not logged in
+    assertTrue(
+        "Dashboard icons must not be present for guest users", dashboardIconsElement.isEmpty());
+
+    // Element holding message displayed when there are no tasks
+    boolean taskResultsMessageDisplayed =
+        wait.until(
+                new Function<WebDriver, WebElement>() {
+                  public WebElement apply(WebDriver driver) {
+                    return driver.findElement(By.id("no-tasks-message"));
+                  }
+                })
+            .isDisplayed();
+    // Message alerting user there are no tasks nearby should be displayed since there are no tasks
+    // yet in the site
+    assertTrue("No tasks in neighborhood message should be displayed", taskResultsMessageDisplayed);
```

Comment:
Is there a way to check whether the correct message is shown on the page? Instead of checking whether there is an element of ID "no-tasks-message", it would be better to check whether the element contains the correct info "Sorry, there are currently no tasks within your neighborhood for you to help with."

Reply:
There's a different div for every message (the other ones are not displayed - set to display:none) so by checking that that one div is displayed we are checking that the correct message is displayed :)

Reply:
Ah I got it.